webapp/src/App.js

```diff
@@ -78,53 +84,63 @@ class DecksList extends React.Component {
 
 class DecksListDeck extends React.Component {
   proptypes = {
-    deck: PropTypes.object,
-    api: PropTypes.instanceOf(Api),
+    deck: PropTypes.object.isRequired,
+    deleteDeck: PropTypes.func.isRequired,
   };
 
+  constructor(props) {
+    super(props);
+
+    this.state = { isDeleted: false };
+  }
+
+  handleDeleteButtonClick() {
+    this.setState({ isDeleted: true });
+  }
+
   render() {
+    const { deck, deleteDeck } = this.props;
+    const { isDeleted } = this.state;
+
     return (
-      //TODO: make the fadeout work
-      <Fade in={true} timeout={1000} exit={true}>
+      <Collapse
+        in={!isDeleted}
+        timeout={500}
+        onExited={() => { deleteDeck(deck.id) }}
```

Comment:
`onExited` is where we can put a callback function to execute once the transition finishes. The `deleteDeck` function that we passed down from `DecksList` does the delete via the API, then updates the `decks` array in its state, causing the `DecksList` to be re-rendered.